actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -144,17 +144,21 @@ def initialize(parameters = {})
     end
 
     # Returns true if another +Parameters+ object contains the same content and
-    # permitted flag, or other Hash-like object contains the same content. This
-    # override is in place so you can perform a comparison with `Hash`.
-    def ==(other_hash)
-      if other_hash.respond_to?(:permitted?)
-        super
+    # permitted flag.
+    def ==(other)
+      if other.respond_to?(:permitted?)
+        self.permitted? == other.permitted? && self.parameters == other.parameters
```

Comment:
If I can't distinguish `ActionController::Parameters.new(a: 1).permit(:a)` and `ActionController::Parameters.new(a: 1)` how should I test that `ActionController::Parameters.new(a: 1).permit(:a)` actually permit's `:a`?

This doesn't make sense to me.